allennlp/service/predictors/predictor.py

```diff
@@ -67,13 +69,16 @@ def _batch_json_to_instances(self, json_dicts: List[JsonDict]) -> List[Tuple[Ins
         return instances
 
     @classmethod
-    def from_archive(cls, archive: Archive, predictor_name: str) -> 'Predictor':
+    def from_archive(cls, archive: Archive, predictor_name: str, config_file=None) -> 'Predictor':
```

Comment:
Why do you need a separate config file here?  The archive already has a config file in it.

Reply:
This was since the archive config used the preprocessed reader, while the demo needs the normal one.

Reply:
Yeah, it's a bit annoying, but I take a trained model, unpack the archive, modify the file, then repack the archive.  It'd probably be easy to write a simple script to do this...